src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -423,4 +433,46 @@ final case class EServicesApiServiceImpl(
       }
     }
   }
+
+  private def getDocumentContentType(document: CatalogManagementDependency.EServiceDoc): Future[ContentType] =
+    ContentType
+      .parse(document.contentType)
+      .fold(
+        ex => Future.failed(ContentTypeParsingError(document.contentType, document.path, ex.map(_.formatPretty))),
+        Future.successful
+      )
+
+  private def convertToMessageEntity(documentDetails: DocumentDetails): MessageEntity = {
+    val randomPath: Path               = Files.createTempDirectory(s"document")
+    val temporaryFilePath: String      = s"${randomPath.toString}/${documentDetails.name}"
+    val file: File                     = new File(temporaryFilePath)
+    val outputStream: FileOutputStream = new FileOutputStream(file)
+    documentDetails.data.writeTo(outputStream)
+    HttpEntity.fromFile(documentDetails.contentType, file)
+  }
```

Comment:
This is a potentially fallible process, let's surround it with a `Try` and call it in the for comprehension in the endpoint
```suggestion
  private def convertToMessageEntity(documentDetails: DocumentDetails): Try[MessageEntity] = Try {
    val randomPath: Path               = Files.createTempDirectory(s"document")
    val temporaryFilePath: String      = s"${randomPath.toString}/${documentDetails.name}"
    val file: File                     = new File(temporaryFilePath)
    val outputStream: FileOutputStream = new FileOutputStream(file)
    documentDetails.data.writeTo(outputStream)
    HttpEntity.fromFile(documentDetails.contentType, file)
  }
```

Reply:
On a second thought, you can use directly `Future` instead of `Try`